src/main/java/org/broadinstitute/hellbender/utils/bwa/BwaMemIndexCache.java

```diff
@@ -0,0 +1,51 @@
+package org.broadinstitute.hellbender.utils.bwa;
+
+import org.apache.spark.api.java.JavaSparkContext;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.utils.Utils;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Manage a global collection of {@link BwaMemIndex} instances.
+ */
+public class BwaMemIndexCache {
+
+    private final static Map<String, BwaMemIndex> instances = new HashMap<>();
+
+    /**
+     * Returns a {@link BwaMemIndex} instance that corresponds to  given index image file.
+     * @param indexImageFile the target image file.
+     * @return never {@code null}.
+     */
+    public static synchronized BwaMemIndex getInstance( final String indexImageFile ) {
+        Utils.nonNull(indexImageFile, "the index image file name provided cannot be null");
+        if (!instances.containsKey(indexImageFile)) {
+            instances.put(indexImageFile, new BwaMemIndex(indexImageFile));
+        }
+        return instances.get(indexImageFile);
+    }
+
+    /**
+     * Closes all instances in the VM.
+     */
+    public static synchronized void closeInstances() {
+        instances.values().forEach(BwaMemIndex::close);
+        instances.clear();
```

Comment:
There is a conceivably recoverable scenario:  we keep a positive reference count on indexes that are currently in use for alignment, and throw an IllegalStateException if we attempt to close an index that's in use.  So later in the program when the index is truly unused you could try again and it would close just fine.  It's also true that no one should be attempting to close an index that may or may not be in use, but I suppose it could happen.